### PR TITLE
Login Modal redesign

### LINKED ISSUE #448: Frontend for SWB authentication
As finalised in https://github.com/publiclab/mapknitter/issues/385, @IshaGupta18 is implementing the front end as 
image
for MK. We can use the same partial files here at SWB as the code should be exactly the same. 
Thanks all

### PR DESCRIPTION
**Pull request by IshaGupta18**
Fixes #448 

Here is the design for the new login modal. I have deliberately kept the username field so that we can discuss if it's a good idea to remove it or not:

image

### CODE REVIEW

**Comment by IshaGupta18** [comment]
@SidharthBansal @jywarren kindly have a look at it and let me know what else can be done here! Thanks!

**Comment by SidharthBansal** [comment]
Well done!!
* [ ] Can you please replace the "login " text with the 
image?
* [ ] Also remove the text box.
* [ ] Please do the changes for /login too.
* [ ] Paste the new screenshots for both modal and /login page.
Thanks

**Comment by SidharthBansal** [comment]
Once you are done with these changes, we can move to further enhancements.
So, please ping me here when done. Thanks

**Comment by IshaGupta18** [comment]
Okay will do! Thanks a lot!

**Comment by IshaGupta18** [comment]
I will remove the text-box, no problem, but then the login button will just be a submit button without a form. Would that be okay?

**Comment by IshaGupta18** [comment]
So I have done the changes and here are the screenshots:

For /login:

image


For the modal:

image

Please have a look and let me know what else can be done here! Thanks!

**Comment by SidharthBansal** [comment]
Well done.
Yeah, we need to remove the text box for sure.
Can you please set remember me as true always and remove it from the
frontend of both modal and page?
Thanks

File: app/views/sessions/login.html.erb [hunk 1]
```diff
@@ -2,21 +2,60 @@
...
+    Create a <a href="http://publiclab.org/signup">Public Lab account</a>
+  </p>
 
   <form id="open_id_form" method="GET" action="/session/new">
 
     <%= hidden_field_tag :authenticity_token, form_authenticity_token -%>
-    <%= hidden_field_tag :back_to,@referer  -%>
-
-    <input tabindex="1" placeholder="username" id="username" name="open_id" type="text" /></p>
-    <input type="hidden" name="return_to" value="<%= params[:return_to] %>" />
```
Comment (SidharthBansal) [inline app/views/sessions/login.html.erb]: May I know why we deleted line 10 and 13?
I think there is need of these lines in the controller

File: app/views/sessions/login.html.erb [hunk 2]
```diff
@@ -2,21 +2,60 @@
...
-
-    <label class="checkbox inline" for="remember_me">Remember</label>
-    <input id="remember_me" name="remember_me" type="checkbox" value="1" />
-    <br>
-    <button tabindex="2" class="btn btn-large btn-primary" type="submit">Log in</button>
+    <%= hidden_field_tag :back_to, (params[:back_to] || request.env['PATH_INFO']) -%>
+    
+    <div>
+      <input type="hidden" name="return_to" value="<%= params[:back_to] || request.env['PATH_INFO'] %>" />
+    </div>
```
Comment (IshaGupta18) [inline app/views/sessions/login.html.erb]: They haven't been deleted, they were over-written by the new code

**Comment by SidharthBansal** [comment]
OK. I will review the lines of code once more after the exam.
In the meantime,
* [ ] please remove the remember me from the frontend and in the backend
set it to one always.
* [ ] place all the 5 buttons together under login.
Thanks

File: app/views/sessions/login.html.erb (hunk 2 above)
Comment (SidharthBansal) [reply app/views/sessions/login.html.erb]: Please see carefully, original line number 10 and 13 is modified to line 19 in new code. Is this necessary?
I think we don't need to modify this line.

**Comment by SidharthBansal** [comment]
- [ ] Also, ``Login with your Public Lab username`` and ``Create a Public lab account`` are not needed now. So please remove them
- [ ] Also, remove both ``OR``s.
I hope that will make frontend much more simpler.

File: app/views/sessions/login.html.erb (hunk 2 above)
Comment (IshaGupta18) [reply app/views/sessions/login.html.erb]: Yes it has been modified, but I didn't do that, they were in the latest pull. Is that not okay?

**Comment by SidharthBansal** [comment]
I don't think they are correct. Make them as master.

**Comment by IshaGupta18** [comment]
Oh okay,no problem, I'll push the changes ASAP!

**Comment by SidharthBansal** [comment]
Oh great!!!
Thanks

**Comment by IshaGupta18** [comment]
Pushed the changes

**Comment by SidharthBansal** [comment]
Please provide the screenshots

**Comment by IshaGupta18** [comment]
Modal
image

Login Page
image

**Comment by SidharthBansal** [comment]
Change the text to
Login via Public Lab
As suggested by Jeff yesterday 
It looks awesome

**Comment by SidharthBansal** [comment]
Can you remove redundancy by using partials properly?

**Comment by IshaGupta18** [comment]
Done the changes!

**Comment by SidharthBansal** [comment]
Please provide the final screenshots

**Comment by SidharthBansal** [comment]
Please provide the final screenshots

**Comment by IshaGupta18** [comment]
image

File: app/views/sessions/login.html.erb (hunk 1 above)
Comment (SidharthBansal) [reply app/views/sessions/login.html.erb]: Can you please ensure no params key is deleted from the views?

**Comment by SidharthBansal** [comment]
Awesome

File: app/views/sessions/login.html.erb (hunk 1 above)
Comment (IshaGupta18) [reply app/views/sessions/login.html.erb]: params[:return_to] is changed to params[:back_to]

**Comment by jywarren** [comment]
Super!!!!

**Comment by SidharthBansal** [comment]
Hi @jywarren this PR needs to be merged with other 3 PRS.
Work is in parallel as I told you yesterday.
Thanks for merging.

**Comment by jywarren** [comment]
Uh oh, sorry, it wasn't marked "do not merge" and it wasn't in our checklist... OK, i'll note to Sebastian that this shouldn't be published until https://github.com/publiclab/plots2/pull/5640 is merged. Thanks!

**Comment by jywarren** [comment]
Er, i meant https://github.com/publiclab/plots2/issues/5652

**Comment by jywarren** [comment]
Ack, sorry! #448 was in the checklist. But it was already checked.

**Comment by SidharthBansal** [comment]
I am really sorry. I did not label it. Sorry.

**Comment by SidharthBansal** [comment]
You did it correctly. We need to `deploy` things simultaneously to production.
I used `merge` in https://github.com/publiclab/spectral-workbench/pull/455#issuecomment-490996340 but I should write `deploy`. Sorry I wrote wrong word.
Thanks for merging.

**Comment by jywarren** [comment]
no issue! I marked it and we should be fine. Thanks for your care and help
here!

**Pull request merged**